### DISCUSSION TITLE
Is there a specific order on how the files are compiled or is there a way where I can specify which files need to be compiled first?

### DISCUSSION THREAD

**Discussion by aniabraham** [swc-project/swc#5888] [Q&A] [answered]
To give some more information, I'm using swc as a transpiler with [ts-node](https://github.com/TypeStrong/ts-node#overview). I am also using mongoose with some additional mongoose plugins in my project. For those who are not familiar with it, certain mongoose plugins need to be loaded before the mongoose models are compiled for it to work correctly and I have added these plugins at the entry point (src/app.js) of my application so that they can be applied globally to all my mongoose models.

When compiling the models using ts-node normally and as well with the [transpile only](https://github.com/TypeStrong/ts-node#transpileonly) option, this seems to work well where the mongoose plugins are loaded first and then the models.

However when using the [swc transpiler](https://github.com/TypeStrong/ts-node#swc) with ts-node it seems that the mongoose models are compiled first and then only is the entry point file compiled. Is there a way where I can specify which files needs to be compiled first or does this need to be a feature request?

**Comment by kdy1** [comment, marked as answer]
`tsc` is simply wrong here. Use `require` if you don't want hoisting of imports, which is the behavior specified by the ES spec.

https://github.com/swc-project/swc/issues/1686
